### ISSUE TITLE
macOS Sierra (10.12) rendering everything black

### ISSUE THREAD

**Issue by ScalaWilliam** [ActionFPS/ActionFPS-Game#101]
Except for white borders around the menu and some ambient lights.

Initial errors:

```
2017-02-03 20:23:09.263184 actionfps[2265:54889] SLPSGetCurrentProcess(): This call is deprecated and should not be called anymore.
2017-02-03 20:23:09.263245 actionfps[2265:54889] set_foreground_operation_state(): This call is deprecated and should not be called anymore.
parsing commandline argument 1: "-NSDocumentRevisionsDebugMode"
parsing commandline argument 2: "YES"
unknown commandline argument: Y
writing to file: clientlog.txt
init: sdl (1.2.14)
2017-02onfps[2265:54889] 4   CarbonCore                          0x00007fff7ea2ea8d Gestalt + 139
2017-02-03 20:23:09.513770 actionfps[2265:54889] 5   SDL                                 0x000000010062cd38 SDL_SoftStretch + 14284
2017-02-03 20:23:09.513802 actionfps[2265:54889] 6   SDL                                 0x0000000100622627 SDL_VideoInit + 531
SDL: compiled version 1.2.14, li-03 20:23:09.473702 actionfps[2265:54889] WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.3 instead of 10.12.3. This is not a bug in Gestalt -- it is a documented limitation. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.
Call location:
2017-02-03 20:23:09.473884 actionfps[2265:54889] 0   CarbonCore                          0x00007fff7eaa11e1 ___Gestalt_SystemVersion_block_invoke + 121
2017-02-03 20:23:09.473924 actionfps[2265:54889] 1   libdispatch.dylib                   0x0000000100670f5c _dispatch_client_callout + 8
2017-02-03 20:23:09.473954 actionfps[2265:54889] 2   libdispatch.dylib                   0x0000000100670f3c dispatch_once_f + 272
2017-02-03 20:23:09.513690 actionfps[2265:54889] 3   CarbonCore                          0x00007fff7ea2f272 _Gestalt_SystemVersion + 976
2017-02-03 20:23:09.513735 actinked version 1.2.15
init: net (1.3.6)
WARNING: unknown commandline argument
init: world (1000)
init: video: sdl
init: video: mode
init: video: misc
init: gl
Renderer: Intel(R) Iris(TM) Graphics 6100 (Intel Inc.)
Driver: 2.1 INTEL-10.22.29
2017-02-03 20:23:10.394369 actionfps[2265:54889] CGBitmapContextInfoCreate: CGColorSpace does't support output
Feb  3 20:23:10  actionfps[2265] <Error>: CGBitmapContextInfoCreate: CGColorSpace does't support output
2017-02-03 20:23:10.394467 actionfps[2265:54889] CGContextDrawImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
Feb  3 20:23:10  actionfps[2265] <Error>: CGContextDrawImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
init: console
```

Followed by constantly repeating:

```
2017-02-03 20:23:10.398124 actionfps[2265:54889] CGBitmapContextInfoCreate: CGColorSpace does't support output
Feb  3 20:23:10  actionfps[2265] <Error>: CGBitmapContextInfoCreate: CGColorSpace does't support output
2017-02-03 20:23:10.398200 actionfps[2265:54889] CGContextDrawImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
Feb  3 20:23:10  actionfps[2265] <Error>: CGContextDrawImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
```

**Comment by ScalaWilliam** [comment]
Stacktrace:
```
2017-02-03 21:04:40.425349 actionfps[2265:27636] CGContextDrawImage: invalid context 0x0. Backtrace:
  <IMG_Load+846>
   <IMG_Load+125>
    <_Z11loadsurfacePKcRiS1_S1_ibbfb+369>
     <_Z11textureloadPKcibbfb+433>
      <_Z13lookuptextureiP7Textureb+249>
       <_ZL18lookupworldtextureib+45>
        <_Z10load_worldPc+5707>
         <_Z13changemapservPciii+241>
          <_Z13parsemessagesiP9playerentR7databufIhEb+2997>
           <_Z14servertoclientiPhib+141>
            <_Z19localservertoclientiPhib+48>
             <_Z10sendpacketiiP11_ENetPacketib+381>
              <_Z11sendwelcomeP6clienti+119>
               <_Z7processP11_ENetPacketii+3884>
                <_Z19localclienttoserveriP11_ENetPacket+31>
                 <_Z16sendpackettoserviP11_ENetPacket+73>
                  <_Z9sendintrov+1106>
                   <_Z13parsemessagesiP9playerentR7databufIhEb+785>
                    <_Z14servertoclientiPhib+141>
                     <_Z19localservertoclientiPhib+48>
                      <_Z10sendpacketiiP11_ENetPacketib+381>
                       <_Z12sendservinfoR6client+218>
                        <_Z12localconnectv+120>
                         <SDL_main+4793>
                          <-[SDLMain applicationDidFinishLaunching:]+73>
                           <__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__+12>
                            <_CFXRegistrationPost+427>
                             <___CFXNotificationPost_block_invoke+50>
                              <-[_CFXNotificationRegistrar find:object:observer:enumerator:]+1827>
                               <_CFXNotificationPost+604>
                                <-[NSNotificationCenter postNotificationName:object:userInfo:]+66>
                                 <-[NSApplication _postDidFinishNotification]+297>
                                  <-[NSApplication _sendFinishLaunchingNotification]+208>
                                   <-[NSApplication(NSAppleEventHandling) _handleAEOpenEvent:]+552>
                                    <-[NSApplication(NSAppleEventHandling) _handleCoreEvent:withReplyEvent:]+661>
                                     <-[NSAppleEventManager dispatchRawAppleEvent:withRawReply:handlerRefCon:]+290>
                                      <_NSAppleEventManagerGenericHandler+102>
                                       <_Z20aeDispatchAppleEventPK6AEDescPS_jPh+544>
                                        <_ZL25dispatchEventAndSendReplyPK6AEDescPS_+39>
                                         <aeProcessAppleEvent+312>
                                          <AEProcessAppleEvent+55>
                                           <_DPSNextEvent+1833>
                                            <-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]+27
                                             <-[NSApplication run]+926>
                                              <CustomApplicationMain+380>
                                               <main+257>
```

**Comment by ScalaWilliam** [comment]
This is precisely down to the `IMG_Load` method which tries to load PNG images for rendering.

**Comment by lucasgautheron** [comment]
Thanks for investigating. What does assaultcube/next has different ?

**Comment by ScalaWilliam** [comment]
Will be fixed in #125.
